mlflow/projects/__init__.py

```diff
@@ -145,7 +147,29 @@ def _run(uri, experiment_id, entry_point="main", version=None, parameters=None,
             work_dir=work_dir, entry_point=entry_point, parameters=parameters,
             experiment_id=experiment_id,
             use_conda=use_conda, storage_dir=storage_dir, run_id=active_run.info.run_id)
-    supported_backends = ["local", "databricks"]
+    elif backend == "kubernetes":
+        from mlflow.projects import kubernetes as kb
+        tracking.MlflowClient().set_tag(active_run.info.run_uuid,
+                                        MLFLOW_PROJECT_ENV,
+                                        "kubernetes")
+        _validate_docker_installation()
+        kube_config = _parse_kubernetes_config(backend_config, work_dir)
+        image = _build_docker_image(work_dir=work_dir,
+                                    image_uri=kube_config["image-uri"],
+                                    base_image=kube_config['base-image'],
```

Comment:
Just curious, why is this different than the image in the `MLProject` file?

Reply:
As docker refers to environment and kubernetes to a backend mode, I thought would be better to let kubernetes config apart from the docker env. Now projects don't need to have a docker_env section if it will run on kubernetes, just the backend_config file.

Reply:
I think this distinction actually make the user workflows more complicated. For example, what happens when I run a MLproject with a `conda_env`?

Instead there are two dimensions here: one is what the project looks like (conda based, docker based), and one is how the project is executed (locally, databricks, k8s). In the future, we can imagine running these docker projects on databricks as well, in that case we don't want to redefine the docker base_image flag for that execution backend for example.